js/snippetPreviewToggler.js

```diff
@@ -0,0 +1,153 @@
+
+var forEach = require( "lodash/forEach" );
+var domManipulation = require( "./helpers/domManipulation.js" );
+
+var previewModes = {
+	desktop: "snippet-editor__view--desktop",
+	mobile: "snippet-editor__view--mobile",
+};
+
+var minimumDesktopWidth = 640;
+
+/**
+ * Constructs the snippet preview toggle.
+ *
+ * @param {string}    previewMode    The default preview mode.
+ * @param {Element[]} previewToggles Array with toggle elements.
+ *
+ * @property {string}   previewMode    The current preview mode.
+ * @propert {Element[]} previewToggles The array with toggle elements.
+ * @constructor
+ */
+var SnippetPreviewToggler = function( previewMode, previewToggles ) {
+	this.previewMode    = previewMode;
+	this.previewToggles = previewToggles;
+};
+
+/**
+ * Initializes the object by setting the current previewMode as the active one.
+ *
+ * @returns {void}
+ */
+SnippetPreviewToggler.prototype.initialize = function() {
+	this._setPreviewMode( this.previewMode, this._findElementByMode( this.previewMode ) );
+};
+
+/**
+ * Sets the events for the preview toggles to switch between the preview modes.
+ *
+ * @returns {void}
+ */
+SnippetPreviewToggler.prototype.setEvents = function() {
+	forEach( this.previewToggles, function( previewToggle ) {
+		previewToggle.addEventListener( "click", function( evt ) {
```

Comment:
Consider moving the anonymous callback into its own function to make this a bit more readable.